commons-shared/src/main/scala/com/avsystem/commons/serialization/GenCodec.scala

```diff
@@ -321,24 +349,24 @@ object GenCodec extends FallbackMapCodecs with TupleGenCodecs {
   implicit val LongCodec: GenCodec[Long] = create(_.readLong(), _ writeLong _)
   implicit val FloatCodec: GenCodec[Float] = create(_.readFloat(), _ writeFloat _)
   implicit val DoubleCodec: GenCodec[Double] = create(_.readDouble(), _ writeDouble _)
-  implicit val BigIntCodec: GenCodec[BigInt] = createNullSafe(i => BigInt(i.readString()), (o, v) => o.writeString(v.toString), allowNull = true)
-  implicit val BigDecimalCodec: GenCodec[BigDecimal] = createNullSafe(i => BigDecimal(i.readString()), (o, v) => o.writeString(v.toString), allowNull = true)
-
-  implicit val JBooleanCodec: GenCodec[JBoolean] = createNullSafe(_.readBoolean(), _ writeBoolean _, allowNull = true)
-  implicit val JCharacterCodec: GenCodec[JCharacter] = createNullSafe(_.readChar(), _ writeChar _, allowNull = true)
-  implicit val JByteCodec: GenCodec[JByte] = createNullSafe(_.readByte(), _ writeByte _, allowNull = true)
-  implicit val JShortCodec: GenCodec[JShort] = createNullSafe(_.readShort(), _ writeShort _, allowNull = true)
-  implicit val JIntegerCodec: GenCodec[JInteger] = createNullSafe(_.readInt(), _ writeInt _, allowNull = true)
-  implicit val JLongCodec: GenCodec[JLong] = createNullSafe(_.readLong(), _ writeLong _, allowNull = true)
-  implicit val JFloatCodec: GenCodec[JFloat] = createNullSafe(_.readFloat(), _ writeFloat _, allowNull = true)
-  implicit val JDoubleCodec: GenCodec[JDouble] = createNullSafe(_.readDouble(), _ writeDouble _, allowNull = true)
-  implicit val JBigIntegerCodec: GenCodec[JBigInteger] = createNullSafe(i => new JBigInteger(i.readString()), (o, v) => o.writeString(v.toString), allowNull = true)
-  implicit val JBigDecimalCodec: GenCodec[JBigDecimal] = createNullSafe(i => new JBigDecimal(i.readString()), (o, v) => o.writeString(v.toString), allowNull = true)
+  implicit lazy val BigIntCodec: GenCodec[BigInt] = createNullSafe(i => BigInt(i.readString()), (o, v) => o.writeString(v.toString), allowNull = true)
```

Comment:
Can you check the impact of this? I'd rather leave them as they are and I would imagine this is not as impactful as `val`s in `CollectionAliases`.

Reply:
So, is it really worth it? This seems to me like a never-ending story - are we going to keep changing all the `val`s to `lazy val`s just for the sake of JS?

I'm really unhappy about the fact that these optimizations are polluting the codebase and disallowing usage of such basic and simple language elements like `val`s.